haystack/nodes/prompt/invocation_layer/hugging_face.py

```diff
@@ -281,8 +262,10 @@ def invoke(self, *args, **kwargs):
             if stream:
                 stream_handler: TokenStreamingHandler = stream_handler or DefaultTokenStreamingHandler()
                 model_input_kwargs["streamer"] = HFTokenStreamingHandler(self.pipe.tokenizer, stream_handler)
-
-            output = self.pipe(prompt, **model_input_kwargs)
+            try:
+                output = self.pipe(prompt, **model_input_kwargs)
+            except Exception as e:
+                raise HuggingFacePipelineInferenceError(f"Text generation failed: {str(e)}") from e
```

Comment:
```suggestion
            try:
                output = self.pipe(prompt, **model_input_kwargs)
            except Exception as e:
                raise HuggingFacePipelineInferenceError("Text generation failed") from e
```

When rereasing an existing Exception, the original Exception will be printed in any case